### ISSUE TITLE
Interop: Mixing CFunctionPtr fails to accept a function argument; crashes the compiler

### ISSUE THREAD

**Issue by rcano** [scala-native/scala-native#841]
This seems to work:
```scala
  val a: CFunctionPtr0[Any] = () => 4
```
This piece of code will crash the compiler:
```scala
  def t(a: => Any): Unit = {
    val b: CFunctionPtr0[Any] = () => a
  }
```
And so does this:
```scala
  def t(a: () => Any): Unit = {
    val b: CFunctionPtr0[Any] = a
  }
```

the stack trace looks like
```
scala.MatchError: native.this.CFunctionPtr.fromFunction0(a) (of class scala.reflect.internal.Trees$ApplyImplicitView)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase.genFunPtrOp(NirCodeGen.scala:1699)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase.genPrimitiveOp(NirCodeGen.scala:1154)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase.genApply(NirCodeGen.scala:1096)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase.genExpr(NirCodeGen.scala:568)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase.genExpr(NirCodeGen.scala:536)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase$$anonfun$35.apply(NirCodeGen.scala:915)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase$$anonfun$35.apply(NirCodeGen.scala:915)
        at scala.scalanative.nir.Focus$$anonfun$sequenced$1.apply(Focus.scala:91)
        at scala.scalanative.nir.Focus$$anonfun$sequenced$1.apply(Focus.scala:90)
        at scala.collection.LinearSeqOptimized$class.foldLeft(LinearSeqOptimized.scala:124)
        at scala.collection.immutable.List.foldLeft(List.scala:84)
        at scala.scalanative.nir.Focus$.sequenced(Focus.scala:90)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase.genBlock(NirCodeGen.scala:915)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase.genExpr(NirCodeGen.scala:621)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase$$anonfun$19.apply(NirCodeGen.scala:515)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase$$anonfun$19.apply(NirCodeGen.scala:515)
        at scala.scalanative.util.ScopedVar$.scoped(ScopedVar.scala:41)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase.genNormalMethodBody(NirCodeGen.scala:514)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase$$anonfun$genMethod$1.apply(NirCodeGen.scala:304)
        at scala.scalanative.util.ScopedVar$.scoped(ScopedVar.scala:41)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase.genMethod(NirCodeGen.scala:279)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase$$anonfun$genMethods$1.apply(NirCodeGen.scala:265)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase$$anonfun$genMethods$1.apply(NirCodeGen.scala:263)
        at scala.collection.immutable.List.foreach(List.scala:392)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase.genMethods(NirCodeGen.scala:263)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase.genNormalClass(NirCodeGen.scala:193)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase$$anonfun$genClass$1.apply(NirCodeGen.scala:166)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase$$anonfun$genClass$1.apply(NirCodeGen.scala:164)
        at scala.scalanative.util.ScopedVar$.scoped(ScopedVar.scala:41)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase.genClass(NirCodeGen.scala:164)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase$$anonfun$apply$4.apply(NirCodeGen.scala:143)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase$$anonfun$apply$4.apply(NirCodeGen.scala:139)
        at scala.collection.immutable.List.foreach(List.scala:392)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase.apply(NirCodeGen.scala:139)
        at scala.tools.nsc.Global$GlobalPhase$$anonfun$applyPhase$1.apply$mcV$sp(Global.scala:467)
        at scala.tools.nsc.Global$GlobalPhase.withCurrentUnit(Global.scala:458)
        at scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:467)
        at scala.tools.nsc.Global$GlobalPhase$$anonfun$run$1.apply(Global.scala:425)
        at scala.tools.nsc.Global$GlobalPhase$$anonfun$run$1.apply(Global.scala:425)
        at scala.collection.Iterator$class.foreach(Iterator.scala:891)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
        at scala.tools.nsc.Global$GlobalPhase.run(Global.scala:425)
        at scala.scalanative.nscplugin.NirCodeGen$NirCodePhase.run(NirCodeGen.scala:115)
        at scala.tools.nsc.Global$Run.compileUnitsInternal(Global.scala:1528)
        at scala.tools.nsc.Global$Run.compileUnits(Global.scala:1513)
        at scala.tools.nsc.Global$Run.compileSources(Global.scala:1508)
        at scala.tools.nsc.Global$Run.compile(Global.scala:1609)
```

**Comment by muxanick** [comment]
very similar to an issue I'm blocked as well #781

**Comment by densh** [comment]
Thanks for the bug report @rcano !

**Comment by densh** [comment]
Fixed in #1426
